### ISSUE TITLE
Req: Link imported files rather than keep copies 

### ISSUE THREAD

**Issue by glocalglocal** [ccbogel/QualCoder#258]
It would be good to be able (optionally?) to link the imported files (docs, audio, video) rather than copy them to the project folders. Consider this scenario: you are adding 10 videos, 1GB each. This means qualcoder keeps 10GB x 6 (inc backups) for original files that will probably be used only once for transcribing. 

I see two difficulties: (a) OSes handle file links differently (b) as ever, linked files may disappear. Not ideal, but one compromise is to backup only the .qda file -- ie keep only one copy of the subfolders to go with the active .qda file.

**Comment by ccbogel** [comment]
Yes, a future option to do.

**Comment by ccbogel** [comment]
@glocalglocal This is a major job. But I am currently working on it.

**Comment by glocalglocal** [comment]
For when linked files are missing, misplaced or replaced, I think qualcoder should store a persistent identifier for each file so that it can tell if you accidentally replace a file with another file with the same name. Eg that identifier could be based on precise length in characters or AV seconds. File size won't do because you can compress audio, video or text to reduce the file size but length as above is less likely to change without the user changing the actual content.

**Comment by ccbogel** [comment]
@glocalglocal Currently it can link to files. I have also made it so that really large files (> 2Gb) are linked and not imported, automatically) As this will assist with REFI-QDA standards. So far I can export REFI-QDA with links (as absolute links re the REFI-QDA standard).
Next I need to text REEFI-QDA imports with absolute and relative links.
A persistent identifier - yes I can do this later (md5 hash). But there is a lot above to resolve and test first.

**Comment by glocalglocal** [comment]
> A persistent identifier - yes I can do this later (md5 hash). But there is a lot above to resolve and test first.

Yes, of course. Won't the md5 hash change if you transcode a media file or change the file type (eg from doc to docx, which is essentially a zip)? The file content (eg number of characters or seconds) wouldn't change, and from the perspective of the user, the two versions may have to be treated as the same material. Perhaps there could be a combination of criteria selected by the user (md5 hash, size, chars/seconds, date, filename etc) but that would complicate coding. 

Not sure if it's relevant but should the link be treated as a URI that could point to a network file, including a web location (eg a youtube video as long as qualcoder/vlc could play it)?

**Comment by ccbogel** [comment]
If you change the file type - you have a different file. The different file type will not be treated as the same file in QualCoder. 

I think using URIs may complicate things a lot. Most web videos are streamed, and it would b quite difficult I think to re-enable a stream in VLC. So - not intendig to do this.

**Comment by ccbogel** [comment]
Links seem to work fine

**Comment by glocalglocal** [comment]
I think there should be an indicator (in addition to the current tooltip) when a file is linked. 

That 'docs:' in the tooltip -- it's not very use what it means.
